File: projects/008-baoyu-design/code/verify_deployment.py
"""Verify the published static files and record HTTP evidence (not UI tests)."""
import hashlib
import json
from pathlib import Path
import sys
from datetime import datetime, timezone
from urllib.request import Request, urlopen

PROJECT = Path(__file__).resolve().parent.parent
ROOT = PROJECT.parents[1]
BASE = 'https://yydshly.github.io/0918_codex_project/'
expected = sys.argv[1]
record = Path(sys.argv[2]) if len(sys.argv) > 2 else PROJECT / 'notes/evidence/deployment.json'
checks = []

def get(path):
    with urlopen(Request(BASE + path, headers={'User-Agent': 'Research-site-verification',
                                              'Cache-Control': 'no-cache'}), timeout=40) as response:
        assert response.status == 200, (path, response.status)
        payload = response.read()
        checks.append({'path': path, 'status': response.status, 'bytes': len(payload)})
        return payload

build = json.loads(get('build.json?verify=' + expected))
assert build['commit'] == expected, (build['commit'], expected)
projects = json.loads((ROOT / 'site-projects.json').read_text(encoding='utf-8'))
assert build['projects'] == [p['slug'] for p in projects]
home = get('').decode('utf-8')
assert '008-baoyu-design/assets/understanding-map.png' in home
for project in projects:
    assert '<html' in get(project['slug'] + '/').decode('utf-8').lower()
prefix = '008-baoyu-design/'
resources = ['app.js', 'styles.css', 'demo/vendor/react.min.js', 'demo/vendor/react-dom.min.js',
             'demo/_ds/reader-kit/_ds_bundle.js', 'demo/_ds/reader-kit/styles.css',
             'deck.html', 'deck-stage.js', 'code/reader-kit/preview.html',
             'notes/understanding.md', 'notes/research.md', 'README.md',
             'THIRD_PARTY_NOTICES.md', 'assets/understanding-map.svg',
             'assets/understanding-map.png', 'demo/downloads/shiye-design.pptx', 'demo/']
responses = {name: get(prefix + name) for name in resources}
assert 'u-summary' in responses['app.js'].decode('utf-8')
hashes = {}
for name in ['assets/understanding-map.png', 'demo/downloads/shiye-design.pptx']:
    local = (PROJECT / name).read_bytes()
    assert local == responses[name], name
    hashes[name] = hashlib.sha256(local).hexdigest()
assert responses['assets/understanding-map.svg'].decode('utf-8').replace('\r\n', '\n') == (
    PROJECT / 'assets/understanding-map.svg').read_text(encoding='utf-8')
result = {'verifiedAt': datetime.now(timezone.utc).isoformat(), 'baseUrl': BASE,
          'projectUrl': BASE + prefix, 'publishedCommit': expected, 'projects': build['projects'],
          'httpChecks': checks, 'matchingBinaryHashes': hashes, 'svgTextMatchesSource': True,
          'limits': 'HTTP/content verification only. Browser interaction evidence is recorded separately.'}
record.write_text(json.dumps(result, ensure_ascii=False, indent=2) + '\n', encoding='utf-8')
print(json.dumps({'commit': expected, 'httpChecks': len(checks), 'projects': len(projects),
                  'binaryAssetsMatch': True, 'record': str(record)}, ensure_ascii=False))

File: projects/010-ai-infra-book/code/verify_deployment.mjs
import assert from 'node:assert/strict';
import {readFile,writeFile} from 'node:fs/promises';
import {createHash} from 'node:crypto';
import {pathToFileURL,fileURLToPath} from 'node:url';
import path from 'node:path';
const project=fileURLToPath(new URL('../',import.meta.url));
const repo=path.resolve(project,'../..');
const base=(process.argv[2]||'https://yydshly.github.io/0918_codex_project/').replace(/\/?$/,'/');
const record=process.argv[3];
const {request}=await import(pathToFileURL(process.env.PLAYWRIGHT_MODULE).href);
const context=await request.newContext({timeout:60000});
const checks=[];
async function get(suffix){const r=await context.get(base+suffix);assert.equal(r.status(),200,suffix);checks.push({path:suffix,status:r.status()});return r}
try{
 const manifest=JSON.parse(await readFile(path.join(repo,'site-projects.json'),'utf8'));
 const build=await (await get('build.json?check='+Date.now())).json();
 if(process.env.EXPECTED_COMMIT)assert.equal(build.commit,process.env.EXPECTED_COMMIT);
 assert.deepEqual(build.projects,[...manifest.map(x=>x.slug)].sort());
 const html=await (await get('')).text();
 const entry=manifest.find(x=>x.slug==='010-ai-infra-book');assert.ok(html.includes(entry.summary));
 for(const item of manifest){assert.ok(html.includes(item.slug+'/'));await get(item.slug+'/')}
 const prefix='010-ai-infra-book/';
 for(const suffix of ['overview.html','styles.css','app.js','overview.js','README.md','demo/README.md','notes/understanding.md','notes/research.md','notes/evidence/upstream.json'])await get(prefix+suffix);
 const source=await (await get(prefix+'index.html')).text();
 assert.ok(source.includes('本地模型显存不足或速度慢、评估多人服务成本，以及微调训练受资源限制'));
 const image=await (await get(prefix+'assets/understanding-map.png')).body();
 const local=await readFile(path.join(project,'assets/understanding-map.png'));
 assert.deepEqual(image,local);
 const result={verifiedAt:new Date().toISOString(),base,commit:build.commit,projects:build.projects,summary:entry.summary,checks,imageSha256:createHash('sha256').update(image).digest('hex'),scope:'核验总站摘要、所有编号入口、010 文档与静态资源、完整理解图字节一致性；交互另见 remote-browser.json。未运行上游模型或 GPU 实验。'};
 if(record)await writeFile(record,JSON.stringify(result,null,2)+'\n');
 console.log(JSON.stringify({passed:true,commit:build.commit,projects:build.projects.length,httpChecks:checks.length,imageSha256:result.imageSha256}));
}finally{await context.dispose()}
